tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -318,12 +319,19 @@ def run_test(self, test):
                 if success:
                     screenshots[i] = screenshot
 
+        # For Mozilla reftest analyzer (included in tbpl):
         log_data = [{"url": nodes[0].url, "screenshot": screenshots[0]}, relation,
                     {"url": nodes[1].url, "screenshot": screenshots[1]}]
+        # For wpt.fyi (included in wptreport.json):
+        new_log_data = {nodes[0].url: (hashes[0], screenshots[0]),
```

Comment:
Instead of passing multiple formats of the same data around, couldn't we just add the hashes key to the existing data, which should be backwards-compatible and rearrange it in the formatter?

Reply:
Right, this is the main question I'd like to ask you: it's currently done this way because of Marionette. If we want to modify the existing data for `reftests_screenshots`, we'd need to modify Marionette as well, as this code path isn't used by Firefox by default. If you'd prefer modifying Marionette, could you give me a pointer to the code?

Reply:
I"m not sure why we need to modify marionette (but sure I"m happy for you to do that). Why can't we just use the same data format and in the formatter (or somewhere earlier like in wpttest.py) compute the hashes if they're missing? That seems like the simplest change to me. What am I not seeing?